### ISSUE TITLE
[Bug]: API example not working - calling psychopy.sound.Sound('A')

### ISSUE THREAD

**Issue by skjerns** [psychopy/psychopy#6765]
### PsychoPy Version

2024.2.0 / Python 3.8

### What OS are your PsychoPy running on?

Windows 10

### Bug Description

In the [API](https://psychopy.org/api/sound/playback.html), the following code is described:

```python
import psychtoolbox as ptb
from psychopy import sound

mySound = sound.Sound('A')
now = ptb.GetSecs()
mySound.play(when=now+0.5)  # play in EXACTLY 0.5s
```

instead of a sound, the following error is thrown:

Instead, the following error is thrown:

```python
>>> import psychtoolbox as ptb
>>> from psychopy import sound
>>>
>>> mySound = sound.Sound('A')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\user\anaconda3\envs\psychopy\lib\site-packages\psychopy_sounddevice\backend_sounddevice.py", line 328, in __init__
    self.setSound(value, secs=self.secs, octave=self.octave,
  File "C:\Users\user\anaconda3\envs\psychopy\lib\site-packages\psychopy_sounddevice\backend_sounddevice.py", line 382, in setSound
    label, s = streams.getStream(sampleRate=self.sampleRate,
TypeError: cannot unpack non-iterable NoneType object
```

Same is happening when switching the sound backend via

```
from psychopy import prefs
prefs.hardware['audioLib'] = ['pyo'] # / 'ptb' / 'pygame'
```

### Expected Behaviour

A sound is played. 

### Steps to Reproduce

install fresh 2024.2.0 from pip, run the following code

```
import psychtoolbox as ptb
from psychopy import sound

mySound = sound.Sound('A')
now = ptb.GetSecs()
mySound.play(when=now+0.5)  # play in EXACTLY 0.5s
```

### Additional context

Python 3.8

**Comment by mscheltienne** [comment]
I am getting a similar error on Ubuntu 24.04 LTS and `2024.2.1` or `main`.
Configuration:

```
$ pactl info
Server String: /run/user/1000/pulse/native
Library Protocol Version: 35
Server Protocol Version: 35
Is Local: yes
Client Index: 66
Tile Size: 65472
User Name: mscheltienne
Host Name: D-1027100163
Server Name: PulseAudio (on PipeWire 1.0.5)
Server Version: 15.0.0
Default Sample Specification: float32le 2ch 48000Hz
Default Channel Map: front-left,front-right
Default Sink: alsa_output.pci-0000_00_1f.3.analog-stereo
Default Source: alsa_output.pci-0000_00_1f.3.analog-stereo.monitor
Cookie: 2f33:63d3
```

------------------------

On `psychopy` main, after fixing a logging error:

```python
from psychopy.sound.backend_ptb import SoundPTB

sound = SoundPTB(value="A", secs=0.5)
```

Traceback:

```
Failure: No such entity
Failure: No such entity
Traceback (most recent call last):
  File "/home/mscheltienne/Downloads/eee.py", line 8, in <module>
    sound = SoundPTB(value="A", secs=0.5)
  File "/home/mscheltienne/git/mscheltienne/psychopy/psychopy/sound/backend_ptb.py", line 337, in __init__
    self.setSound(value, secs=self.secs, octave=self.octave,
  File "/home/mscheltienne/git/mscheltienne/psychopy/psychopy/sound/backend_ptb.py", line 426, in setSound
    _SoundBase.setSound(self, value, secs, octave, hamming, log)
  File "/home/mscheltienne/git/mscheltienne/psychopy/psychopy/sound/_base.py", line 194, in setSound
    self._setSndFromNote(value.capitalize(), secs, octave,
  File "/home/mscheltienne/git/mscheltienne/psychopy/psychopy/sound/_base.py", line 250, in _setSndFromNote
    self._setSndFromFreq(thisFreq, secs, hamming=hamming)
  File "/home/mscheltienne/git/mscheltienne/psychopy/psychopy/sound/_base.py", line 266, in _setSndFromFreq
    self._setSndFromArray(outArr)
  File "/home/mscheltienne/git/mscheltienne/psychopy/psychopy/sound/backend_ptb.py", line 500, in _setSndFromArray
    self.track = audio.Slave(self.stream.handle, data=self.sndArr,
  File "/home/mscheltienne/git/mscheltienne/psychopy/psychopy/sound/backend_ptb.py", line 617, in stream
    label, s = devices[self.speaker.index].getStream(
TypeError: cannot unpack non-iterable NoneType object
0.6547  ERROR   No speaker device found with index 'default'
Failure: No such entity
Failure: No such entity
```

Listed devices:

```python
from psychopy.sound.backend_ptb import getDevices

devices = getDevices(kind="output")
```

Output:

```
Failure: No such entity
Failure: No such entity
{'HDA Intel PCH: ALC3246 Analog (hw:0,0)': {'DeviceIndex': 0.0, 'HostAudioAPIId': 8.0, 'HostAudioAPIName': 'ALSA', 'DeviceName': 'HDA Intel PCH: ALC3246 Analog (hw:0,0)', 'NrInputChannels': 2.0, 'NrOutputChannels': 2.0, 'LowInputLatency': 0.008707482993197279, 'HighInputLatency': 0.034829931972789115, 'LowOutputLatency': 0.008707482993197279, 'HighOutputLatency': 0.034829931972789115, 'DefaultSampleRate': 44100.0, 'id': 0}, 'HDA Intel PCH: HDMI 0 (hw:0,3)': {'DeviceIndex': 2.0, 'HostAudioAPIId': 8.0, 'HostAudioAPIName': 'ALSA', 'DeviceName': 'HDA Intel PCH: HDMI 0 (hw:0,3)', 'NrInputChannels': 0.0, 'NrOutputChannels': 2.0, 'LowInputLatency': -1.0, 'HighInputLatency': -1.0, 'LowOutputLatency': 0.008707482993197279, 'HighOutputLatency': 0.034829931972789115, 'DefaultSampleRate': 44100.0, 'id': 2}, 'HDA Intel PCH: HDMI 1 (hw:0,7)': {'DeviceIndex': 3.0, 'HostAudioAPIId': 8.0, 'HostAudioAPIName': 'ALSA', 'DeviceName': 'HDA Intel PCH: HDMI 1 (hw:0,7)', 'NrInputChannels': 0.0, 'NrOutputChannels': 8.0, 'LowInputLatency': -1.0, 'HighInputLatency': -1.0, 'LowOutputLatency': 0.008707482993197279, 'HighOutputLatency': 0.034829931972789115, 'DefaultSampleRate': 44100.0, 'id': 3}, 'HDA Intel PCH: HDMI 2 (hw:0,8)': {'DeviceIndex': 4.0, 'HostAudioAPIId': 8.0, 'HostAudioAPIName': 'ALSA', 'DeviceName': 'HDA Intel PCH: HDMI 2 (hw:0,8)', 'NrInputChannels': 0.0, 'NrOutputChannels': 8.0, 'LowInputLatency': -1.0, 'HighInputLatency': -1.0, 'LowOutputLatency': 0.008707482993197279, 'HighOutputLatency': 0.034829931972789115, 'DefaultSampleRate': 44100.0, 'id': 4}, 'HDA Intel PCH: HDMI 3 (hw:0,9)': {'DeviceIndex': 5.0, 'HostAudioAPIId': 8.0, 'HostAudioAPIName': 'ALSA', 'DeviceName': 'HDA Intel PCH: HDMI 3 (hw:0,9)', 'NrInputChannels': 0.0, 'NrOutputChannels': 8.0, 'LowInputLatency': -1.0, 'HighInputLatency': -1.0, 'LowOutputLatency': 0.008707482993197279, 'HighOutputLatency': 0.034829931972789115, 'DefaultSampleRate': 44100.0, 'id': 5}, 'sysdefault': {'DeviceIndex': 6.0, 'HostAudioAPIId': 8.0, 'HostAudioAPIName': 'ALSA', 'DeviceName': 'sysdefault', 'NrInputChannels': 128.0, 'NrOutputChannels': 128.0, 'LowInputLatency': 0.008707482993197279, 'HighInputLatency': 0.034829931972789115, 'LowOutputLatency': 0.008707482993197279, 'HighOutputLatency': 0.034829931972789115, 'DefaultSampleRate': 44100.0, 'id': 6}}
Failure: No such entity
Failure: No such entity
```

**Comment by mscheltienne** [comment]
And 2024.1.4 works.

**Comment by peircej** [comment]
That's very interesting. @mdcutone I wonder if you can replicate this and track it down. If it's changed since 2024.1 then it must be a relatively small amount of code that's changed

**Comment by TEParsons** [comment]
I believe this is that, if `Sound` doesn't get given a value for `stereo`, it uses -1, which seems to fail in some cases. I think this is fixed by #6867, as it'll mean that the `SpeakerDevice` knows whether it's stereo or mono so this can be retrieved by Sound if not otherwise supplied.

The reason this flew under the radar is that Builder always supplies a value for `stereo`, so it had only hit users writing code directly.
